packages/esm-appointments-app/src/form/appointments-form.component.tsx

```diff
@@ -483,7 +490,7 @@ const AppointmentsForm: React.FC<AppointmentsFormProps> = ({
             <Controller
               name="selectedService"
               control={control}
-              render={({ field: { onBlur, onChange, value, ref } }) => (
+              render={({ field: { onBlur, onChange, value, ref }, fieldState }) => (
                 <Select
                   id="service"
                   invalidText="Required"
```

Comment:
And then I'd do this for all the fields that should be required:

```tsx
    invalid={!!fieldState?.error?.message}
    invalidText={fieldState.error?.message}
```

Reply:
More on this approach here https://o3-docs.openmrs.org/docs/recipes/validate-forms-using-rhf-and-zod